RoboJackets/igvc-software
# Scan To Pointcloud Node Rewrite

# Description

Rewrite of the scan to pointcloud node.

This PR does the following:
- Converts scan_to_pointcloud into a C++ class.
- Splits scan_to_pointcloud into a .cpp and .h file.
- Adds unit tests to verify functionality of the node.

Fixes #741

# Testing steps
## Test Case 1
1. Run `catkin_make run_tests_igvc_platform_rostest_src_tests_test_test_scan_to_pointcloud.test`

Expectation: All tests pass

## Test Case 2
1. Run `roscore`
2. Run `roslaunch igvc_platform lidar.launch`
3. Run `rostopic echo /pc2`
4. Run 
```
rostopic pub /scan sensor_msgs/LaserScan "header:
  seq: 0
  stamp: {secs: 0, nsecs: 0}
  frame_id: ''
angle_min: 0.0
angle_max: 0.0
angle_increment: 0.0
time_increment: 0.0
scan_time: 0.0
range_min: 0.0
range_max: 0.0
ranges: [0]
intensities: [0]" 
```

Expectation: `rostopic echo /pc2` should return
```
header: 
  seq: 1
  stamp: 
    secs: 0
    nsecs:         0
  frame_id: "/lidar"
height: 1
width: 0
fields: 
  - 
    name: "x"
    offset: 0
    datatype: 7
    count: 1
  - 
    name: "y"
    offset: 4
    datatype: 7
    count: 1
  - 
    name: "z"
    offset: 8
    datatype: 7
    count: 1
is_bigendian: False
point_step: 16
row_step: 0
data: []
is_dense: False
---
``` 


# Self Checklist
